Sources/Citadel/SFTP/SFTPServerInboundHandler.swift

```
@@ -0,0 +1,173 @@
+                            message: "",
+                            languageTag: "EN"
+                        )
+                    )
+                )
+            }.whenFailure { _ in
+                context.channel.close(promise: nil)
+            }
+        case .handle(let command):
+            print(command)
+        case .status(let command):
+            print(command)
+        case .data(let command):
+            print(command)
+        case .mkdir(let command):
+            print(command)
+        case .stat(let command):
+            let promise = context.eventLoop.makePromise(of: SFTPFileAttributes.self)
+            promise.completeWithTask {
+                try await self.delegate.fileAttributes(atPath: command.path)
+            }
+            
+            promise.futureResult.whenSuccess { attributes in
+                context.writeAndFlush(
+                    NIOAny(SFTPMessage.attributes(
+                        .init(
+                            requestId: command.requestId,
+                            attributes: attributes
+                        )
+                    )),
+                    promise: nil
+                )
+            }
+        case .lstat(let command):
+            let promise = context.eventLoop.makePromise(of: SFTPFileAttributes.self)
+            promise.completeWithTask {
+                try await self.delegate.fileAttributes(atPath: command.path)
+            }
+                
+            promise.futureResult.whenSuccess { attributes in
+                context.writeAndFlush(
+                    NIOAny(SFTPMessage.attributes(
+                        .init(
+                            requestId: command.requestId,
+                            attributes: attributes
+                        )
+                    )),
+                    promise: nil
+                )
+            }
+        case .attributes(let command):
+            print(command)
```

same here for print statement

---

this is a server-response type, so it cannot be received by a server

---

This should throw an error